src/main/java/org/broadinstitute/consent/http/resources/DatasetResource.java

```diff
@@ -365,6 +366,20 @@ public Response datasetAutocomplete(@Auth AuthUser authUser, @PathParam("partial
         }
     }
 
+    @GET
+    @Path("/search")
+    @Produces("application/json")
+    @PermitAll
+    public Response searchDatasets(@Auth AuthUser authUser, @QueryParam("query") String query){
+        try {
+            User user = userService.findUserByEmail(authUser.getEmail());
+            List<Dataset> datasets = datasetService.searchDatasets(query, user);
+            return Response.ok(datasets, MediaType.APPLICATION_JSON).build();
```

Comment:
The common approach we use is to use the entity method for the object returned. This is a little cleaner in that you don't have to specify the media type.
```suggestion
            return Response.ok().entity(datasets).build();
```